### ISSUE TITLE
`RangeError: Maximum call stack size exceeded` when using `.extend()` method

### ISSUE THREAD

**Issue by thawankeane** [sam-goodwin/zod-class#21]
The following code is throwing an compile error `RangeError: Maximum call stack size exceeded` 

```ts
class User extends Z.class({
  name: z.string(),
}) {}

class UserAccount extends User.extend({
  email: z.string().email(),
}) {}
```

Here is my `package.json` file with each dependency version

```json
{
  "name": "zod-class-example",
  "version": "0.0.1",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsup",
    "dev": "tsup --watch"
  },
  "dependencies": {
    "zod": "^3.22.4",
    "zod-class": "^0.0.12"
  },
  "devDependencies": {
    "@types/node": "^20.11.4",
    "tsup": "^8.0.1",
    "typescript": "^5.3.3"
  }
}
```
I tried to build with `tsup` and `tsc`, both with the same error

**Comment by sam-goodwin** [comment]
Thanks for reporting. Will take a look when I next get some time. Sorry for late reply, been busy!

**Comment by sam-goodwin** [comment]
Am digging into this. It's a stack overflow error in the typescript compiler caused by the `StaticProperties` step in `extend`.

**Comment by sam-goodwin** [comment]
Should be fixed in `0.0.13` - sorry that it took me so long to fix this. LMK if it's working on your end!
